### PR TITLE
Replace Travis CI with GitHub Actions

### PR DESCRIPTION
**Pull request by jonkoops**
Replaces Travis CI (which has been broken since the their [domain migration](https://docs.travis-ci.com/user/migrate/open-source-repository-migration)) with GitHub Actions. This has the added benefit that forks are also able to run the CI without additional setup work.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    branches: [master]
+  pull_request:
+    branches: [master]
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21]
```
Comment (jonkoops) [inline .github/workflows/main.yml:13]: I've had to omit versions 6 and 7 here, which we are technically compatible with, but cannot enable due to development dependencies that are not compatible with these Node.js versions.

File: .travis.yml [hunk 2]
```diff
@@ -1,17 +0,0 @@
...
-sudo: false
-before_install:
-  - export "PATH=./node_modules/.bin:$PATH"
-install: npm install
-script:
-  - gulp lint
-  - gulp dist
-  - gulp test karma
-after_success:
-  - gulp coveralls
```
Comment (jonkoops) [inline .travis.yml:16]: I've not included `coveralls` in the GitHub version, as this seems to error out on an invalid file. Likely this broke at some point, but was not caught as CI was broken. We'll have to look at this at a later point to get it fixed.

File: package-lock.json
Comment (jonkoops) [inline package-lock.json:1]: I've checked in the lockfile, as [recommended by NPM](https://docs.npmjs.com/cli/v10/configuring-npm/package-lock-json#description). This should help prevent breaking dependency bumps from occurring by accident.

Comment (puleos) [reply package-lock.json:1]: 👍

**Review by puleos — APPROVED** [review]

**Pull request merged**